### DISCUSSION TITLE
CMake build support

### DISCUSSION THREAD

**Discussion by grahamboree** [recastnavigation/recastnavigation#568] [Polls]
The current CMake scripts for Recast have some issues (see #524 #472 #441)

I have zero experience with CMake (and zero desire to learn it, frankly) so I'm really unfit to maintain Recast's CMake build system.  I also feel that keeping multiple build systems for Recast working and compatible in parallel is going to be a tedious task at best.  

I know there are some people who are really into CMake out there, so if you're one of those people and are willing to champion the effort to maintain the cmake scripts, I'd be very thankful!

**Comment by elsid** [comment]
CMake support is added to recastnavigation so it could be integrated into [OpenMW](https://gitlab.com/OpenMW/openmw). Removing its support would create problems for OpenMW.

I'm not sure why you suggest to remove CMake support just because of few minor issues. But you can always ask bug reporters to contribute to the project considering some of them even know possible solutions. It's an open source project after all. Looking at these issues I see that https://github.com/recastnavigation/recastnavigation/issues/472 is fixed, https://github.com/recastnavigation/recastnavigation/issues/441 issue is not really an issue and https://github.com/recastnavigation/recastnavigation/issues/524 seems to be specific to the reporter setup which requires more input to understand the problem and provide a solution.

**Reply by grahamboree** [reply]
Apologies, I should have been clearer.  I'm not suggesting we remove CMake support.  I've raised this as a concern because I wanted to identify a clear advocate or group of community members who can support the CMake pipeline.  Seems like you're passionate about this, which is great!  This all comes from my efforts to work through our issue and PR backlog and make sure that problems don't languish.  If there's a group of folks like yourself in the community who can address CMake-related issues that crop up, I'm more than happy to support that.  I just can't be that person :)
